Sudoku/sudoku.js

```diff
@@ -1,18 +1,181 @@
-//
-// This is only a SKELETON file for the "Sudoku" exercise. It's been provided as a
-// convenience to get you started writing code faster.
-//
+const {buildMap} = require('./buildMap')
+const NUMBERS = [1,2,3,4,5,6,7,8,9]
 
-var Sudoku = function(board) {
-  
-};
+const checkNumbersNeededAreUsedForGivenContainerType = (type, board) => { // TODO: HELP! What do I name this function to be informative AND concise?
+  const boardContainerName = {
+    square: 'sqrs',
+    row: 'rows',
+    column: 'cols'
+  }
+  const slotContainerName = {
+    square: 'sqr',
+    row: 'row',
+    column: 'col'
+  }
+  const parentContainer = board[boardContainerName[type]]
+  for( let container in parentContainer ) {
+    container = parseInt(container)
+    let numbersNotUsed = board.numbersNeeded[type][container].slice()
+    // console.log('NUMBERSNOTUSED', numbersNotUsed[0] === board.potentialNumbers[0].numbers[0])
+    const numbersUsedInContainer = board.potentialNumbers.filter( n => {
+      return parseInt(n[slotContainerName[type]]) === container
+    })
+    for( let numberSet of numbersUsedInContainer ) {
+      numberSet.numbers.forEach( n => {
+        numbersNotUsed = removeNumbers(n, numbersNotUsed)
+      })
+    }
+    if(numbersNotUsed.length > 0){
+      const err = {
+        fromWho: type,
+        error: 'ERROR: Bad Board'
+      }
+      throw err
+    }
+  }
+}
 
-Sudoku.prototype.isSolved = function() {
+const arrayIntersect = (arr1, arr2) => {
+  return arr1.reduce(
+    (r,a) => arr2.includes(a) && r.concat(a) || r, [] // What does this mean?!
```

Comment:
I believe this means that the arrays will either be concatenated  or it will be empty space that will eventually throw an error.